h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoost.java

```diff
@@ -215,6 +215,9 @@
       error("_grow_policy", "must use tree_method=hist for grow_policy=lossguide");
 
     if ((_train != null) && (_parms._monotone_constraints != null)) {
+      if (_parms._tree_method == XGBoostModel.XGBoostParameters.TreeMethod.approx) {
```

Comment:
I would just future proof this code by adding an assert that the set of possible tree methods is the current one. Then we can catch that a new tree method was added and that we might need to revisit this condition.